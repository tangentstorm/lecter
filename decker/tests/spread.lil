
# test double-dot "spread" indexing

# dictionaries
d1:("a","b")dict "one","two"
d2:("a","b")dict "three","four"
d3:("a","b")dict "five","six"
ds:(list d1),(list d2),(list d3)
show[ds..a]              # extract a column
show[ds..a:"new"]        # amend-each dict
show[ds]                 # spreading assignments don't modify the source object;
ds:ds..a:"new" show[ds]  # ...unless you have an explicit leading assign.

# lists
ls:(list 11,22,33,44),(list 55,66,77,88)
show[ls.[2]]      # extract a column

# lists-of-tables
t1:table "ABC" cross "DE"
t2:table "MN" cross "Q"
ts:(list t1),(list t2)
show[ts..c1]      # extract columns
show[ts..c1[1]]   # extract cell from each column

# tables
show[t1.[1]] # extract the first row

# nesting
show[((range 3)..a:99)..b:32]

# old versions of lil accidentally shadowed a local 'x':
local x:5
show[ds..a:x]
