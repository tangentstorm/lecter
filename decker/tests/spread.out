("one","three","five")
({"a":"new","b":"two"},{"a":"new","b":"four"},{"a":"new","b":"six"})
({"a":"one","b":"two"},{"a":"three","b":"four"},{"a":"five","b":"six"})
({"a":"new","b":"two"},{"a":"new","b":"four"},{"a":"new","b":"six"})
(33,77)
(("D","D","D","E","E","E"),("Q","Q"))
("D","Q")
("B","D")
({"a":99,"b":32},{"a":99,"b":32},{"a":99,"b":32})
({"a":5,"b":"two"},{"a":5,"b":"four"},{"a":5,"b":"six"})
